src/PairedRouting.php

```diff
@@ -772,6 +773,37 @@ public function add_permalink_settings_notice() {
 		<?php
 	}
 
+	/**
+	 * Update canonical URL for legacy reader paired urls.
+	 *
+	 * @param string $redirect_url Canonical redirect URL.
+	 *
+	 * @return string Updated canonical URL.
+	 */
+	public function maybe_update_paired_url( $redirect_url ) {
+
+		global $wp_rewrite;
+
+		/**
+		 * To address issue AMP version of a comment page.
+		 * Reference: https://github.com/ampproject/amp-wp/issues/6355
+		 */
+		if ( ! empty( $redirect_url ) &&
+			! empty( $wp_rewrite->comments_pagination_base ) &&
+			AMP_Options_Manager::get_option( Option::PAIRED_URL_STRUCTURE ) === Option::PAIRED_URL_STRUCTURE_LEGACY_READER
+		) {
+
+			/**
+			 * Reference: https://regex101.com/r/544758/1/
+			 */
+			$regex        = "#/({$wp_rewrite->comments_pagination_base}-[0-9]+?)/amp/\\1(/+)?($|\?)#";
```

Comment:
@dhaval-parekh can we please use `amp_get_slug()` instead of `/amp/ `
 for example : https://github.com/ampproject/amp-wp/blob/develop/src/PairedRouting.php#L730-L739